SQL/tgstation_schema.sql

```diff
@@ -255,7 +255,8 @@ CREATE TABLE `library` (
   KEY `deleted_idx` (`deleted`),
   KEY `idx_lib_id_del` (`id`,`deleted`),
   KEY `idx_lib_del_title` (`deleted`,`title`),
-  KEY `idx_lib_search` (`deleted`,`author`,`title`,`category`)
+  KEY `idx_lib_search` (`deleted`,`author`,`title`,`category`),
+  KEY `idx_admin_lib_search` (`author`,`title`,`category`)
```

Comment:
Thanks. Being honest I'm not really familiar with SQL keys (or I wasn't before), my assumption was it let us tell the DB that this form of request would be more common and thus worth caching.

I added book_id support later in dev and uh, forgot to update this. It is a decently rare case but in a perfect world it would be nice to cover for it.

Modifying all uses of the lib_search pattern would not be hard, how much is the overhead to adding a new index?

Reply:
Actually it doesn't matter since all these search queries have a `LIKE` that starts with a wildcard and that prevents an index from being used. Apparently a covering index (one that contains all columns in both `SELECT` and `WHERE`) can work with a starting wildcard but I'd have to test that.

Otherwise as is `idx_lib_search` and `idx_admin_lib_search` won't do anything useful due to the addition of `book_id`.

Reply:
wouldn't both indexes be useless then? since that pattern is only used when wildcarding since it's for searches and such

Reply:
I've done some fairy rough testing and basically don't worry about it; regardless of index design it seems like the query optimizer decides using the primary key or `idx_deleted` is faster. I'm not sure if this is because a covering index doesn't work with starting wildcards as claimed by a single comment on stackoverflow or the scan space is too small for it to matter.

W/e either way I'd just get rid of this new index, it's not needed.

Reply:
aight, yeeted